bolt-design-system/bolt
# Add accordion component

## Jira

- http://vjira2:8080/browse/BDS-1256

## Summary

Add accordion component with demos, docs and tests.

## Details

The Accordion component was initially based on the existing Dropdown component and reworked to meet the requirements of an accordion. Some of this work could eventually be ported back to improve the Dropdown component in the future.

It uses the [Handorgel](https://github.com/oncode/handorgel) accessible Accordion plugin, extended to work with the HTML structure we are using.

It consists to two new custom elements: `bolt-accordion` ([twig](https://github.com/bolt-design-system/bolt/blob/feature/accordion/packages/components/bolt-accordion/src/accordion.twig)) and `bolt-accordion-item` ([twig](https://github.com/bolt-design-system/bolt/blob/feature/accordion/packages/components/bolt-accordion/src/AccordionItem/accordion-item.twig)).

It works without JS with only minor functional differences. Mainly, you can have multiple sections open at a time.

See accordion schema and demos for implementation details.

## How to test
- Review the [schema](https://feature-accordion.boltdesignsystem.com/pattern-lab/patterns/02-components-accordion/index.html#schema)
- Cross-browser test the [demos](https://feature-accordion.boltdesignsystem.com/pattern-lab/patterns/02-components-accordion/index.html)
- Review related bug fixes:
  - Fixed schema docs error when nested props have multiple types ([commit](https://github.com/bolt-design-system/bolt/pull/1159/commits/700b07645ed8a9709eaca738c24fb2c7dbf4b109))
  - Polyfill 'remove()' to fix IE11 errors in accordion ([commit](https://github.com/bolt-design-system/bolt/pull/1159/commits/c8564910a67ecb85e3cfd74836854eb7b1d35132))
  - Add check for 'link' to analytics JS, throwing error in accordion ([commit](https://github.com/bolt-design-system/bolt/pull/1159/commits/a746a9e91bb04b130b93b1f1df8305a6f0b3d9e4))
  - Prevent mutation observer from re-rendering when using light dom ([commit](https://github.com/bolt-design-system/bolt/pull/1159/commits/f30c99dc26a10b6abaea8ae55eeae6546403cd7e))
